src/jit/codegenxarch.cpp

```diff
@@ -188,7 +224,13 @@ void                CodeGen::genEmitGSCookieCheck(bool pushReg)
         if (compiler->lvaKeepAliveAndReportThis() && compiler->lvaTable[compiler->info.compThisArg].lvRegister &&
             (compiler->lvaTable[compiler->info.compThisArg].lvRegNum == REG_ECX))
         {
```

Comment:
Actually there is no trashing of RDX here by accident on Linux.  Since thisArg on Linux will be passed in REG_EDI the if check will always be false.  Hence gs check was using RCX which is REG_ARG3.  Since there is no bug in emitting GS cookie check please update the title of PR and also the bug filed.